### PR TITLE
split vrack allowed services calls

### PR DESCRIPTION
**Pull request by blemoineovh**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12446
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description
After changing the endpoint vrack-allowed-service to be able to recieve calls for only one family, this PR adds the ability to get rid of timeout when requesting the services of all the families available to add to a vrack.
This is done by calling each family separately in the manager and show a counter of the calls.
The getting of the service id and a display name is processed just like before with a more fail safe approach.

## Related
See the changes in the 2api endpoint vrack-allowed-services, it is still able to retrieve the call to all available services just like before, without any family filter.

### CODE REVIEW

File: packages/manager/modules/vrack/src/dashboard/vrack.html [hunk 1]
```diff
@@ -179,7 +179,7 @@ <h3 data-translate="vrack_description_title"></h3>
                 </p>
             </div>
         </section>
-
+        <!---->
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.html:182]: I imagine that the comment should include the name of the section

**Review by ghyenne — CHANGES_REQUESTED** [review]

File: packages/manager/modules/vrack/src/partials/available.html [hunk 2]
```diff
@@ -1,89 +1,148 @@
...
+                    'ovh-font ovh-font-dedicatedCloud': serviceType === 'dedicatedCloud' || serviceType === 'dedicatedCloudDatacenter',
+                    'oui-icon oui-icon-cloud-essential_concept': serviceType === 'managedBareMetal',
+                    'ovh-font ovh-font-public-cloud': serviceType === 'cloudProject',
+                    'ovh-font ovh-font-iplb' : serviceType === 'ipLoadbalancing',
+                    'ovh-font ovh-font-ip': serviceType === 'ip',
+                    'ovh-font ovh-font-server': serviceType === 'dedicatedServer' || serviceType === 'dedicatedServerInterface',
+                    'ovh-font ovh-font-network': serviceType === 'dedicatedConnect',
+                    'ovh-font ovh-font-vRack': serviceType === 'legacyVrack'}"
+                >
+                </i>
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html:61]: agreed, but it was like that before, shouldn't we avoid to make refactoring in general if the ticket does not says we should refacto ?

File: packages/manager/modules/vrack/src/dashboard/vrack.html
Comment (anooparveti) [inline packages/manager/modules/vrack/src/dashboard/vrack.html:1]: Unnecessary changes in this file and all this file can be discarded from the PR

File: packages/manager/modules/vrack/src/dashboard/vrack.less [hunk 3]
```diff
@@ -104,4 +104,8 @@
   &__empty-description-text {
     margin: 2.1875rem 0 0.9375rem 0;
   }
+
+  .vrack__service-fetching {
+    float: right;
+  }
```
Comment (anooparveti) [inline packages/manager/modules/vrack/src/dashboard/vrack.less:110]: **nitpick:** Use `float-right` bootstrap class instead.

File: packages/manager/tools/dev-server-config/src/proxy/aapi/index.js
Comment (anooparveti) [inline packages/manager/tools/dev-server-config/src/proxy/aapi/index.js:1]: This change should not be included in the PR

File: packages/manager/tools/webpack-dev-server/src/config.js
Comment (anooparveti) [inline packages/manager/tools/webpack-dev-server/src/config.js:1]: This change should not be included in the PR?

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -69,6 +69,7 @@
     "@types/node": "^18.0.0",
     "@typescript-eslint/eslint-plugin": "^4.27.0",
     "@typescript-eslint/parser": "^4.27.0",
+    "angular-mocks": "*",
```
Comment (anooparveti) [inline package.json:72]: **question:** What is the need for this package?

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js [hunk 5]
```diff
@@ -49,11 +49,61 @@ export const STATUS = {
...
+  cloudProject: 'not_fetched',
+  dedicatedCloud: 'not_fetched',
+  dedicatedCloudDatacenter: 'not_fetched',
+  dedicatedConnect: 'not_fetched',
+  dedicatedServer: 'not_fetched',
+  dedicatedServerInterface: 'not_fetched',
+  ip: 'not_fetched',
+  ipLoadbalancing: 'not_fetched',
+  legacyVrack: 'not_fetched',
+  ovhCloudConnect: 'not_fetched',
```
Comment (anooparveti) [inline packages/manager/modules/vrack/src/dashboard/vrack.constant.js:79]: **nitpick:** Loaders should be boolean. Maintaining them as non-standardised strings can make it hard to maintain

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (anooparveti) [inline yarn.lock:1]: I dont see the need to include any packages for the required changes. Please discard the changes on this file

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/vrack/src/partials/mapped.html [hunk 6]
```diff
@@ -71,15 +70,15 @@ <h2 class="vrack__section-title" data-translate="vrack_your_vrack"></h2>
                 <ul class="vrack__service-list vrack__service-list_group">
                     <li
                         class="vrack__service vrack__service_selectable d-flex"
-                        data-ng-click="$ctrl.toggleDeleteService(serviceType, service.id)"
                         data-ng-class="{ vrack__service_selected: $ctrl.isSelected(serviceType, service.id), vrack__service_pending: $ctrl.isPending(service.id) || $ctrl.loaders.adding || $ctrl.loaders.deleting }"
+                        data-ng-click="$ctrl.toggleDeleteService(serviceType, service.id)"
                     >
                         <span
-                            data-translate="vrack_private_cloud_vm_network_label"
+                            data-translate="'vrack_private_cloud_vm_network_label'"
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/mapped.html]: ok done

File: packages/manager/modules/vrack/src/partials/available.html [hunk 7]
```diff
@@ -1,89 +1,148 @@
...
                 <span
-                    class="vrack__service-group-name"
-                    data-ng-click="$ctrl.toggleAccordion('available', 'dedicatedCloudNetworks', $index)"
+                    class="loading"
+                    data-ng-if="$ctrl.loaders.services[serviceType] == 'fetching'"
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: added strict equal

File: packages/manager/modules/vrack/src/partials/available.html [hunk 8]
```diff
@@ -1,89 +1,148 @@
...
-            <li
-                class="vrack__service d-flex"
-                data-ng-repeat="service in services | orderBy: service.niceName track by service.id"
-                data-ng-click="$ctrl.toggleAddService(service.trueServiceType, service.id)"
-                data-ng-class="{ vrack__service_selected: $ctrl.isSelected(service.trueServiceType, service.id), vrack__service_pending: $ctrl.isPending(service.id) || $ctrl.loaders.adding || $ctrl.loaders.deleting }"
+        </li>
+        <li class="vrack__section-type">
+            <span
+                class="loading"
+                data-ng-if="$ctrl.loaders.services[serviceType] == 'fetching'"
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: added strict equal

File: packages/manager/modules/vrack/src/partials/available.html [hunk 9]
```diff
@@ -1,89 +1,148 @@
...
+        <!-- we hide the dedicatedServerInterface because they are sent in the dedicated server group -->
+        <li
+            class="vrack__service-type vrack__service_type_{{serviceType}}"
+            data-ng-init="$ctrl.setAccordionState('available', 'types', $index, true)"
+            data-ng-repeat="(serviceType, services) in $ctrl.data.allowedServices track by $index"
+            data-ng-if="
+                (
+                  $ctrl.loaders.services[serviceType] == 'fetching'
+                ||
+                $ctrl.loaders.services[serviceType] == 'loaded'
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: added strict equal

File: packages/manager/modules/vrack/src/dashboard/vrack.html (hunk 1 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.html]: yup, finally it is cleaner without comment

File: packages/manager/modules/vrack/src/dashboard/vrack.less (hunk 3 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.less]: added float-right instead of this specific css

File: packages/manager/tools/webpack-dev-server/src/config.js
Comment (blemoineovh) [reply packages/manager/tools/webpack-dev-server/src/config.js:1]: removed the hack to access local 2api

File: package.json (hunk 4 above)
Comment (blemoineovh) [reply package.json]: it was to run unit testing with mocks. i removed this

File: yarn.lock
Comment (blemoineovh) [reply yarn.lock:1]: done

File: packages/manager/tools/dev-server-config/src/proxy/aapi/index.js
Comment (blemoineovh) [reply packages/manager/tools/dev-server-config/src/proxy/aapi/index.js:1]: ok removed

File: packages/manager/modules/vrack/src/partials/available.html
Comment (anooparveti) [inline packages/manager/modules/vrack/src/partials/available.html:1]: **nitpick:** `$ctrl.loaders.services[serviceType]` has only 2 values `fetching` and `loaded`. So please make it a boolean to check if the services are loaded. Using custom strings is not maintainable.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 10]
```diff
@@ -65,6 +69,219 @@ export default class VrackMoveDialogCtrl {
     this.vrackService = CucVrackService;
     this.atInternet = atInternet;
     this.changeOwnerTrackLabel = `${VRACK_DASHBOARD_TRACKING_PREFIX}::change-owner`;
+    this.servicesFamilies = SERVICE_FAMILIES_LIST;
```
Comment (anooparveti) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:72]:
```suggestion
    this.SERVICE_FAMILIES_LIST = SERVICE_FAMILIES_LIST;
```

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js (hunk 5 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.constant.js]: changed too booleans

File: packages/manager/modules/vrack/src/partials/available.html [hunk 11]
```diff
@@ -1,89 +1,148 @@
...
-                        data-ng-click="$ctrl.toggleAddService(serviceType, service.id)"
-                        data-ng-class="{ vrack__service_selected: $ctrl.isSelected(serviceType, service.id), vrack__service_pending: $ctrl.isPending(service.id) || $ctrl.loaders.adding || $ctrl.loaders.deleting }"
+            <!-- grouped services -->
+            <ul
+                class="vrack__service-list vrack__service_type_grouped_{{serviceType}}"
+                data-ng-if="!$ctrl.vrackService.isGroupedServiceType(serviceType)"
+            >
+                <li
+                    class="vrack__service d-flex"
+                    data-ng-if="serviceType === 'ip'"
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: added constant in template, imported in controller from constants file

File: packages/manager/modules/vrack/src/partials/available.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: added mapping function to make this logic outside of the template

File: packages/manager/modules/vrack/src/dashboard/vrack.html
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.html:1]: removed changes in this file

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 12]
```diff
@@ -909,144 +1066,131 @@ export default class VrackMoveDialogCtrl {
...
+    const mappedServices = mapValues(
+      allServices,
+      (serviceList, serviceType) => {
+        const mappedstuff = this.constructor.fillServiceData(
+          serviceType,
+          serviceList,
+        );
+        return mappedstuff;
+      },
     );
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:1134]:
```suggestion
    const mappedServices = Object.values(allServices).map((serviceList, serviceType) => this.constructor.fillServiceData(
          serviceType,
          serviceList,
        )
    );
```

File: packages/manager/modules/cloud-universe-components/src/vrack/service.js [hunk 13]
```diff
@@ -25,13 +25,7 @@ export default class CucVrackService {
...
 
   static getGroupedServiceTypes() {
-    return [
-      'dedicatedCloudDatacenter',
-      'managedBareMetalDatacenter',
-      'dedicatedCloud',
-      'managedBareMetal',
-      'dedicatedServerInterface',
-    ];
+    return GROUPED_SERVICES_TYPES;
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/cloud-universe-components/src/vrack/service.js:28]: i did not modify this, it was made like this by @cbourgois

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 14]
```diff
@@ -909,144 +1066,131 @@ export default class VrackMoveDialogCtrl {
     return this.form.serviceToMove !== null && !this.loaders.moving;
   }
 
-  static hasServices(services) {
-    return keys(services).length > 0;
+  setUserRelatedContent() {
+    this.OvhApiMe.v6()
+      .get()
+      .$promise.then((user) => {
+        if (user.ovhSubsidiary === 'FR') {
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js:1073]: it is used to modify this.vRackCloudRoadmapGuide but it does not seem to be used anywhere, can i remove it ?

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 15]
```diff
@@ -909,144 +1066,131 @@ export default class VrackMoveDialogCtrl {
...
+
+    /**
+     * for each service, make a call and append the result to a literal object
+     * this allows to see progressively the services for each families in the picker
+     * and get rid of the timeout if there are too many services fetched at once
+     */
+    this.SERVICE_FAMILIES_LIST.forEach((serviceFamily) => {
+      this.loaders.services[serviceFamily] = 'fetching';
+
+      const self = this;
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js:1096]: ok updated

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 16]
```diff
@@ -65,6 +71,220 @@ export default class VrackMoveDialogCtrl {
...
+      case 'ip':
+        if (typeof service === 'string') {
+          formattedService = {
+            id: service,
+            niceName: service,
+            trueServiceType: 'ip',
+          };
+        } else {
+          formattedService = service;
+        }
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js]: split in function done

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 17]
```diff
@@ -65,6 +71,220 @@ export default class VrackMoveDialogCtrl {
...
+        dedicatedCloudDatacenter.niceName = last(service.datacenter.split('_'));
+        if (isObject(service.dedicatedCloud)) {
+          dedicatedCloudDatacenter.dedicatedCloud.niceName = `${service.dedicatedCloud.serviceName} (${service.dedicatedCloud.description})`;
+        } else {
+          dedicatedCloudDatacenter.dedicatedCloud = {
+            niceName: service.dedicatedCloud,
+          };
+        }
+        dedicatedCloudDatacenter.trueServiceType = 'dedicatedCloudDatacenter';
+        formattedService = dedicatedCloudDatacenter;
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js]: split in function done

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js]: updated with coreconfig

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js [hunk 18]
```diff
@@ -49,11 +49,85 @@ export const STATUS = {
...
+    'ovh-font ovh-font-public-cloud': serviceType === 'cloudProject',
+    'ovh-font ovh-font-iplb': serviceType === 'ipLoadbalancing',
+    'ovh-font ovh-font-ip': serviceType === 'ip',
+    'ovh-font ovh-font-server':
+      serviceType === 'dedicatedServer' ||
+      serviceType === 'dedicatedServerInterface',
+    'ovh-font ovh-font-network': serviceType === 'dedicatedConnect',
+    'ovh-font ovh-font-vRack': serviceType === 'legacyVrack',
+  };
+}
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.constant.js:121]: It shouldn't to have a function in constant file

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 19]
```diff
@@ -65,6 +72,228 @@ export default class VrackMoveDialogCtrl {
...
+    return formattedService;
+  }
+
+  static getDedicatedServerInterfaceNiceName(service) {
+    const formattedService = VrackMoveDialogCtrl.getDedicatedServerNiceName(
+      service.dedicatedServer,
+    );
+    formattedService.id = service.dedicatedServerInterface;
+    formattedService.niceName = `${formattedService.niceName} - ${service.name}`;
+    formattedService.trueServiceType = 'dedicatedServerInterface';
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:159]: You should move in constant file "dedicatedServerInterface". the same for all constant for trueServiceType

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 20]
```diff
@@ -190,92 +421,27 @@ export default class VrackMoveDialogCtrl {
...
         allServices = mapValues(allServices, (servicesParams, serviceType) => {
           let services = servicesParams;
-          if (isArray(services)) {
-            services = map(services, (service) =>
-              this.constructor.fillServiceData(serviceType, service),
-            );
+          if (Array.isArray(services) && services.length) {
+            services = map(services, (service) => {
+              this.constructor.fillServiceData(serviceType, service);
+            });
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:444]:
```suggestion
            services = services.map((service) => 
              this.constructor.fillServiceData(serviceType, service)
            );
```

File: packages/manager/tools/webpack-dev-server/src/config.js [hunk 21]
```diff
@@ -18,10 +18,6 @@ module.exports = (env) => {
   const isContainer = env.container || process.env.CONTAINER;
 
   const sso = new Sso(region, envConfig);
-
-  if (yn(env.local2API) || yn(process.env.npm_package_config_local2API)) {
-    proxy.unshift(serverProxy.aapi);
-  }
```
Comment (ghyenne) [inline packages/manager/tools/webpack-dev-server/src/config.js:24]: Why you remove the global config ?

File: packages/manager/tools/dev-server-config/src/proxy/aapi/index.js [hunk 22]
```diff
@@ -5,7 +5,7 @@ const config = {
 };
 
 module.exports = {
-  context: ['/engine/2api'],
+  context: ['/engine/2api/vrack'],
```
Comment (ghyenne) [inline packages/manager/tools/dev-server-config/src/proxy/aapi/index.js:8]: You shouldn't change this config

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js (hunk 18 above)
Comment (anooparveti) [inline packages/manager/modules/vrack/src/dashboard/vrack.constant.js:121]: **nitpick:** Can you also add it in default export?

File: packages/manager/modules/vrack/src/partials/available.html [hunk 23]
```diff
@@ -1,89 +1,138 @@
...
-                'ovh-font ovh-font-public-cloud': serviceType === 'cloudProject',
-                'ovh-font ovh-font-iplb' : serviceType === 'ipLoadbalancing',
-                'ovh-font ovh-font-ip': serviceType === 'ip',
-                'ovh-font ovh-font-server': serviceType === 'dedicatedServer' || serviceType === 'dedicatedServerInterface',
-                'ovh-font ovh-font-network': serviceType === 'dedicatedConnect',
-                'ovh-font ovh-font-vRack': serviceType === 'legacyVrack'}"
+            <oui-spinner data-size="s"></oui-spinner>
+            <span class="oui-badge oui-badge_info"
+                >{{ $ctrl.getAllowedServicesCounter }}/{{
+                $ctrl.SERVICE_FAMILIES_LIST.length }}</span
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/partials/available.html:13]:
```suggestion
            <span class="oui-badge oui-badge_info"
                data-ng-bind="{{ $ctrl.getAllowedServicesCounter + '/' + $ctrl.SERVICE_FAMILIES_LIST.length }}"
                ></span
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js (hunk 18 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.constant.js]: added to exports

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js [hunk 24]
```diff
@@ -26,13 +26,28 @@ export const VRACK_URLS = {
...
+export const IP_SERVICE = 'ip';
+export const IPLB_SERVICE = 'ipLoadbalancing';
+export const DEDICATED_SERVICE = 'dedicatedServer';
+export const DSI_SERVICE = 'dedicatedServerInterface';
+export const DCLOUDDC_SERVICE = 'dedicatedCloudDatacenter';
+export const DCLOUD_SERVICE = 'dedicatedCloud';
+export const OVHCC_SERVICE = 'ovhCloudConnect';
+export const CP_SERVICE = 'cloudProject';
+export const LVRACK_SERVICE = 'legacyVrack';
+export const MBM_SERVICE = 'managedBareMetal';
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.constant.js:42]: It would be more efficient if you use only one object.

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js [hunk 25]
```diff
@@ -49,11 +64,84 @@ export const STATUS = {
...
+      serviceType === 'managedBareMetal',
+    'ovh-font ovh-font-public-cloud': serviceType === 'cloudProject',
+    'ovh-font ovh-font-iplb': serviceType === 'ipLoadbalancing',
+    'ovh-font ovh-font-ip': serviceType === IP_SERVICE,
+    'ovh-font ovh-font-server':
+      serviceType === 'dedicatedServer' || serviceType === DSI_SERVICE,
+    'ovh-font ovh-font-network': serviceType === 'dedicatedConnect',
+    'ovh-font ovh-font-vRack': serviceType === 'legacyVrack',
+  };
+}
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.constant.js:134]: Why you create many constants above if you don't reuse here.

File: packages/manager/modules/vrack/src/partials/available.html [hunk 26]
```diff
@@ -1,89 +1,138 @@
...
-                'ovh-font ovh-font-vRack': serviceType === 'legacyVrack'}"
+            <oui-spinner data-size="s"></oui-spinner>
+            <span
+                class="oui-badge oui-badge_info"
+                data-ng-bind="{{ $ctrl.getAllowedServicesCounter + '/' + $ctrl.SERVICE_FAMILIES_LIST.length }}"
+            ></span>
+        </li>
+        <li class="vrack__section-type">
+            <span
+                class="loading"
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/partials/available.html:18]: You are sure this class (loading) exist ?

**Review by ghyenne — CHANGES_REQUESTED** [review]

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js (hunk 19 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js]: updated to constant

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js (hunk 20 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js]: added your suggestion

File: packages/manager/tools/dev-server-config/src/proxy/aapi/index.js (hunk 22 above)
Comment (blemoineovh) [reply packages/manager/tools/dev-server-config/src/proxy/aapi/index.js]: removed

File: packages/manager/modules/vrack/src/partials/available.html (hunk 23 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: done

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js [hunk 27]
```diff
@@ -26,13 +26,44 @@ export const VRACK_URLS = {
...
+  DC_SERVICE,
+  DEDICATED_SERVICE,
+  DSI_SERVICE,
+  IPLB_SERVICE,
+  IP_SERVICE,
+  LVRACK_SERVICE,
+  MBMDC_SERVICE,
+  MBM_SERVICE,
+  OVHCC_SERVICE,
+};
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.constant.js:58]: Are you sure you need the both ? It's best to keep the object i think

File: packages/manager/modules/vrack/src/partials/available.html [hunk 28]
```diff
@@ -1,89 +1,148 @@
...
-                            data-translate="vrack_private_cloud_vm_network_label"
-                        ></span>
-                        <oui-spinner
-                            data-size="s"
-                            data-ng-if="$ctrl.isPending(service.id)"
+                    </oui-spinner>
+                </li>
+                <li
+                    class="vrack__service d-flex"
+                    data-ng-if="serviceType !== 'ip'"
```
[the remark this answers is not in the document]
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: done

File: packages/manager/tools/webpack-dev-server/src/config.js (hunk 21 above)
Comment (blemoineovh) [reply packages/manager/tools/webpack-dev-server/src/config.js]: restored

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js (hunk 24 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.constant.js]: made it inside one object

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js (hunk 25 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.constant.js]: reused

File: packages/manager/modules/vrack/src/partials/available.html (hunk 26 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: it is not, i removed this

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js (hunk 27 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.constant.js:58]: i made the object using the constants, and changed the controller to use the object who gather the constants instead of importing each constant separately

File: packages/manager/modules/vrack/src/partials/available.html (hunk 26 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/partials/available.html]: removed

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js (hunk 18 above)
Comment (blemoineovh) [reply packages/manager/modules/vrack/src/dashboard/vrack.constant.js]: moved to controller

File: packages/manager/modules/vrack/src/partials/available.html [hunk 29]
```diff
@@ -1,89 +1,133 @@
...
+                        data-size="s"
+                    >
+                    </oui-spinner>
+                </li>
+            </ul>
+            <!--            END: grouped-->
+            <!--            dedicated cloud or managed bare metal -->
+            <ul
+                class="vrack__service-group-list vrack__service_type_{{serviceType}}"
+                data-ng-if="serviceType === 'dedicatedCloud' || serviceType === 'managedBareMetal'"
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/partials/available.html:97]:
```suggestion
                data-ng-if="[$ctrl.SERVICE_NAMES.DCLOUD_SERVICE, $ctrl.SERVICE_NAMES.MBM_SERVICE].includes(serviceType)"
```

**Comment by ghyenne** [comment]
The check git (lint code / lint commit) are failed. Can you check ? The ref of ticket are missing for example in commit message.

File: packages/manager/modules/vrack/src/dashboard/vrack.constant.js [hunk 30]
```diff
@@ -49,11 +68,86 @@ export const STATUS = {
...
+    'ovh-font ovh-font-dedicatedCloud':
+      serviceType === SERVICES_NAMES.DCLOUD_SERVICE ||
+      serviceType === SERVICES_NAMES.DCLOUDDC_SERVICE,
+    'oui-icon oui-icon-cloud-essential_concept':
+      serviceType === SERVICES_NAMES.MBM_SERVICE,
+    'ovh-font ovh-font-public-cloud': serviceType === SERVICES_NAMES.CP_SERVICE,
+    'ovh-font ovh-font-iplb': serviceType === SERVICES_NAMES.IPLB_SERVICE,
+    'ovh-font ovh-font-ip': serviceType === SERVICES_NAMES.IP_SERVICE,
+    'ovh-font ovh-font-server':
+      serviceType === 'dedicatedServer' ||
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.constant.js:134]:
```suggestion
      serviceType === SERVICES_NAMES.DEDICATED_SERVICE ||
```

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 31]
```diff
@@ -65,6 +72,228 @@ export default class VrackMoveDialogCtrl {
...
+
+  /**
+   * make a niceName with the product description
+   * @param service
+   */
+  static getOvhCloudConnectNiceName(service) {
+    return {
+      id: service.uuid,
+      niceName: service.product,
+      trueServiceType: SERVICES_NAMES['OVHCC_SERVICE'],
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:135]:
```suggestion
      trueServiceType: SERVICES_NAMES.OVHCC_SERVICE,
```

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 32]
```diff
@@ -65,6 +72,228 @@ export default class VrackMoveDialogCtrl {
...
+    const formattedService = Object.create(service);
+    formattedService.id = service.name;
+    // by default the reverse seem to be equal to the name,
+    // so do not display redundant information.
+    if (service.reverse && service.reverse !== service.name) {
+      formattedService.niceName = `${service.name} (${service.reverse})`;
+    } else {
+      formattedService.niceName = service.name;
+    }
+    formattedService.trueServiceType = SERVICES_NAMES['DEDICATED_SERVICE'];
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:149]:
```suggestion
    formattedService.trueServiceType = SERVICES_NAMES.DEDICATED_SERVICE;
```

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 33]
```diff
@@ -65,6 +72,228 @@ export default class VrackMoveDialogCtrl {
...
+    return formattedService;
+  }
+
+  static getDedicatedServerInterfaceNiceName(service) {
+    const formattedService = VrackMoveDialogCtrl.getDedicatedServerNiceName(
+      service.dedicatedServer,
+    );
+    formattedService.id = service[SERVICES_NAMES['DSI_SERVICE']];
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:157]:
```suggestion
    formattedService.id = service[SERVICES_NAMES.DSI_SERVICE];
```

Comment (ghyenne) [reply packages/manager/modules/vrack/src/dashboard/vrack.controller.js:157]: And same for the all in same contoller

File: packages/manager/modules/vrack/src/dashboard/vrack.controller.js [hunk 34]
```diff
@@ -190,103 +421,38 @@ export default class VrackMoveDialogCtrl {
...
+          if (Array.isArray(services) && services.length) {
+            services = services.map((service) =>
               this.constructor.fillServiceData(serviceType, service),
             );
           }
           return services;
         });
 
-        if (has(allServices, 'dedicatedCloud')) {
+        if (has(allServices, SERVICES_NAMES['DCLOUD_SERVICE'])) {
```
Comment (ghyenne) [inline packages/manager/modules/vrack/src/dashboard/vrack.controller.js:449]:
```suggestion
        if (has(allServices, SERVICES_NAMES.DCLOUD_SERVICE)) {
```
and same for other in same conttroller

**Review by ghyenne — CHANGES_REQUESTED** [review]
